src/Program/Model/City.hs

```diff
@@ -70,17 +70,21 @@ import Control.Monad.Writer.Strict
 
 -- | Map of all city objects (buildings, roads, etc).
 data City = City
-    { activeObjId       :: !Int
---    , activeObjSnapshot :: !(Maybe LocatedCityObject)
-    , objectsIn         :: !CityObjectCollection
-    , cityTransform     :: !(GLfloat, Vector2 GLfloat)
-    , ground            :: !CityGround
-    , csettings         :: !CitySettings
-    , clutter           :: !(LS.MultiLineString 3 GLfloat, WiredGeometry)
-    , buildingColors    :: !(Maybe (PS.PointArray 4 GLfloat))
-    --, drawTextures      :: !Bool
-    , originLonLatAlt   :: !(Maybe (Vector 3 GLfloat))
-    , srid              :: !(Maybe Int)
+    { activeObjId        :: !Int
+--    , activeObjSnapshot  :: !(Maybe LocatedCityObject)
+    , objectsIn          :: !CityObjectCollection
+    , cityTransform      :: !(GLfloat, Vector2 GLfloat)
+    , ground             :: !CityGround
+    , csettings          :: !CitySettings
+    , clutter            :: !(LS.MultiLineString 3 GLfloat, WiredGeometry)
+    , buildingColors     :: !(Maybe (PS.PointArray 4 GLfloat))
+    --, drawTextures       :: !Bool
+    , originLonLatAlt    :: !(Maybe (Vector 3 GLfloat))
+    , srid               :: !(Maybe Int)
+    , defaultBlockColor  :: !(Maybe JSString)
+    , defaultActiveColor :: !(Maybe JSString)
+    , defaultStaticColor :: !(Maybe JSString)
+    , defaultLineColor   :: !(Maybe JSString)
```

Comment:
The same story: too many properties duplicated in different data types.
Also color properties can be grouped together in a single data type. Then you can use this type in City, parsedFeatureCollection, and every single Building too.